File: src/org/zeroxlab/momome/impl/DetailActivity.java
/*
 * Authored By Julian Chu <walkingice@0xlab.org>
 *
 * Copyright (c) 2012 0xlab.org - http://0xlab.org/
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.zeroxlab.momome.impl;

import org.zeroxlab.momome.R;
import org.zeroxlab.momome.Momo;
import org.zeroxlab.momome.MomoApp;
import org.zeroxlab.momome.MomoModel;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.TextView;
import android.widget.Toast;
import android.view.KeyEvent;
import android.view.WindowManager;

public class DetailActivity extends Activity implements Momo {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        TextView text = (TextView)findViewById(R.id.detail_text);
        Intent intent = getIntent();
        String data = intent.getStringExtra(CROSS_ENTRY_DATA_KEY);
        String comment = intent.getStringExtra(CROSS_ENTRY_COMMENT_KEY);

        text.setText(data + ":" + comment);
    }

    @Override
    public void onAttachedToWindow() {
        this.getWindow().setType(WindowManager.LayoutParams.TYPE_KEYGUARD);
        super.onAttachedToWindow();
    }

     @Override
     public boolean onKeyDown(int keyCode, KeyEvent event) {
         if (keyCode == KeyEvent.KEYCODE_HOME) {
             finish();
         }
         return super.onKeyDown(keyCode, event);
     }
}

File: app/src/main/java/org/zeroxlab/momome/impl/DetailActivity.kt
/*
 * Authored By Julian Chu <walkingice@0xlab.org>
 *
 * Copyright (c) 2012 0xlab.org - http://0xlab.org/
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.zeroxlab.momome.impl

import android.app.Activity
import android.os.Bundle
import android.widget.TextView
import org.zeroxlab.momome.Momo
import org.zeroxlab.momome.R
import java.text.SimpleDateFormat
import java.util.*

class DetailActivity : Activity(), Momo {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val viewData = findViewById(R.id.detail_data) as TextView
        val viewComment = findViewById(R.id.detail_comment) as TextView
        val viewTime = findViewById(R.id.detail_time) as TextView

        val intent = intent
        val data = intent.getStringExtra(Momo.CROSS_ENTRY_DATA_KEY)
        val comment = intent.getStringExtra(Momo.CROSS_ENTRY_COMMENT_KEY)
        val time = intent.getLongExtra(Momo.CROSS_ENTRY_TIME_KEY, System.currentTimeMillis())

        val date = Date(time)
        viewData.text = data
        viewComment.text = comment
        viewTime.text = sSDF.format(date)
    }

    companion object {
        protected val sSDF = SimpleDateFormat("MMM dd,yyyy HH:mm:ss")
    }
}
